pkg/domain/infra/abi/images.go

```diff
@@ -336,20 +336,54 @@ func (ir *ImageEngine) Push(ctx context.Context, source string, destination stri
 		pushOptions.Writer = os.Stderr
 	}
 
-	pushedManifestBytes, pushError := ir.Libpod.LibimageRuntime().Push(ctx, source, destination, pushOptions)
-	if pushError == nil {
-		if options.DigestFile != "" {
-			manifestDigest, err := manifest.Digest(pushedManifestBytes)
-			if err != nil {
-				return err
+	pushImage := func(dest string) error {
+		pushedManifestBytes, pushError := ir.Libpod.LibimageRuntime().Push(ctx, source, dest, pushOptions)
+		if pushError == nil {
+			if options.DigestFile != "" {
+				manifestDigest, err := manifest.Digest(pushedManifestBytes)
+				if err != nil {
+					return err
+				}
+
+				if err := os.WriteFile(options.DigestFile, []byte(manifestDigest.String()), 0644); err != nil {
+					return err
+				}
 			}
+		}
+		return pushError
+	}
 
-			if err := os.WriteFile(options.DigestFile, []byte(manifestDigest.String()), 0644); err != nil {
-				return err
+	var pushError error
+	if options.AllTags {
+		// retrieves all of the images that are associated with the source given by the user
+		listOptions := libimage.ListImagesOptions{}
+		listOptions.Filters = []string{fmt.Sprintf("reference=%s", source)}
+		images, err := ir.Libpod.LibimageRuntime().ListImages(ctx, []string{}, &listOptions)
+		if err != nil {
+			return err
+		}
+
+		for _, image := range images {
+			if len(image.Names()) < 1 {
+				return errors.New("attempting to push an unnamed image")
+			}
+
+			tagPrefixIdx := strings.LastIndex(image.Names()[0], ":")
```

Comment:
So if I have an image tagged with (repo1:tag1, repo2:tag2, repo3:tag3), and do `podman push --all-tags repo2`, this will push to `repo2:tag1`??